File: suc/swift/044.swift
import SwiftUI

struct AppTokens {
    struct Colors {
        static let primary = Color(red: 1.0, green: 0.44, blue: 0.26)
        static let secondary = Color(red: 1.0, green: 0.72, blue: 0.3)
        static let tertiary = Color(red: 1.0, green: 0.82, blue: 0.5)
        static let background = Color(red: 1.0, green: 0.95, blue: 0.88)
        static let surface = Color(red: 1.0, green: 1.0, blue: 1.0)
        static let surfaceVariant = Color(red: 1.0, green: 0.88, blue: 0.7)
        static let outline = Color(red: 0.84, green: 0.8, blue: 0.78)
        static let success = Color(red: 0.4, green: 0.73, blue: 0.42)
        static let warning = Color(red: 1.0, green: 0.79, blue: 0.16)
        static let error = Color(red: 0.83, green: 0.18, blue: 0.18)
        static let onPrimary = Color(red: 1.0, green: 1.0, blue: 1.0)
        static let onSecondary = Color(red: 0.24, green: 0.15, blue: 0.14)
        static let onBackground = Color(red: 0.24, green: 0.15, blue: 0.14)
        static let onSurface = Color(red: 0.24, green: 0.15, blue: 0.14)
    }

    struct TypographyTokens {
        static let display = Font.system(size: 28, weight: .bold)
        static let title = Font.system(size: 18, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular)
        static let label = Font.system(size: 12, weight: .medium)
    }

    struct Shapes {
        static let small: CGFloat = 6.0
        static let medium: CGFloat = 10.0
        static let large: CGFloat = 16.0
    }

    struct Spacing {
        static let sm: CGFloat = 6.0
        static let md: CGFloat = 10.0
        static let lg: CGFloat = 16.0
        static let xl: CGFloat = 24.0
    }
}

struct CustomSlider: View {
    @Binding var value: Float
    let activeColor: Color
    let inactiveColor: Color
    let thumbColor: Color
    let trackHeight: CGFloat = 10
    let thumbDiameter: CGFloat = 20

    var body: some View {
        GeometryReader { geometry in
            ZStack(alignment: .leading) {
                Capsule()
                    .fill(inactiveColor)
                    .frame(height: trackHeight)

                Capsule()
                    .fill(activeColor)
                    .frame(width: geometry.size.width * CGFloat(value), height: trackHeight)

                Circle()
                    .fill(thumbColor)
                    .frame(width: thumbDiameter, height: thumbDiameter)
                    .offset(x: geometry.size.width * CGFloat(value) - thumbDiameter / 2)
                    .gesture(
                        DragGesture()
                            .onChanged { gesture in
                                let dragLocation = gesture.location.x
                                let newValue = Float(min(max(0, dragLocation / geometry.size.width), 1))
                                self.value = newValue
                            }
                    )
            }
        }
        .frame(height: thumbDiameter)
    }
}

struct RootScreen: View {
    @State private var progress: Float = 0.42 // 与原图42%进度保持一致

    var body: some View {
        ZStack {
            // 背景色铺满整个屏幕，无额外边距
            AppTokens.Colors.background.ignoresSafeArea(.all)

            // 主内容容器：移除顶部额外留白，与原 Kotlin Column 布局对齐
            VStack(spacing: AppTokens.Spacing.lg) {
                // 标题：仅跟随容器统一内边距，无额外顶部间距
                Text("Order Tracking")
                    .font(AppTokens.TypographyTokens.display)
                    .foregroundColor(AppTokens.Colors.onBackground)
                    .frame(maxWidth: .infinity, alignment: .leading)

                // 订单信息卡片（与原 Kotlin Card 布局一致）
                VStack(spacing: AppTokens.Spacing.md) {
                    Text("Order #25491")
                        .font(AppTokens.TypographyTokens.title)
                        .foregroundColor(AppTokens.Colors.onSurface)
                        .frame(maxWidth: .infinity, alignment: .leading)

                    Text("Estimated Delivery: 2 days")
                        .font(AppTokens.TypographyTokens.body)
                        .foregroundColor(AppTokens.Colors.secondary)
                        .frame(maxWidth: .infinity, alignment: .leading)

                    // 进度条（与原 Kotlin LinearProgressIndicator 尺寸一致）
                    GeometryReader { geometry in
                        ZStack(alignment: .leading) {
                            Capsule()
                                .fill(AppTokens.Colors.surfaceVariant)
                                .frame(height: 10)

                            Capsule()
                                .fill(AppTokens.Colors.primary)
                                .frame(width: geometry.size.width * CGFloat(progress), height: 10)
                        }
                    }
                    .frame(height: 10)

                    // 进度文本（左右对齐，与原图一致）
                    HStack {
                        Text("Processing")
                            .font(AppTokens.TypographyTokens.label)
                            .foregroundColor(AppTokens.Colors.secondary)

                        Spacer() // 用 Spacer 确保左右对齐，比 frame(maxWidth) 更精准

                        Text("\(Int(progress * 100))%")
                            .font(AppTokens.TypographyTokens.label)
                            .foregroundColor(AppTokens.Colors.primary)
                    }
                }
                .padding(AppTokens.Spacing.lg)
                .background(AppTokens.Colors.surface)
                .cornerRadius(AppTokens.Shapes.large)
                .shadow(
                    color: Color.black.opacity(0.15),
                    radius: 8,
                    x: 0,
                    y: 4 // 与原 Kotlin ElevationMapping.level2 阴影参数一致
                )

                // 进度调整区域（与原 Kotlin Box 布局一致）
                VStack(spacing: AppTokens.Spacing.md) {
                    Text("Adjust Progress")
                        .font(AppTokens.TypographyTokens.title)
                        .foregroundColor(AppTokens.Colors.onSurface)
                        .frame(maxWidth: .infinity, alignment: .center)

                    CustomSlider(
                        value: $progress,
                        activeColor: AppTokens.Colors.primary,
                        inactiveColor: AppTokens.Colors.surfaceVariant,
                        thumbColor: AppTokens.Colors.primary
                    )

                    Button(action: {
                        progress = 1.0 // 点击后进度设为100%（已送达）
                    }) {
                        Text("Mark as Delivered")
                            .font(AppTokens.TypographyTokens.label)
                            .foregroundColor(AppTokens.Colors.onPrimary)
                            .padding(.vertical, AppTokens.Spacing.md)
                            .frame(maxWidth: .infinity)
                            .background(AppTokens.Colors.primary)
                            .cornerRadius(AppTokens.Shapes.medium)
                    }
                    .buttonStyle(PlainButtonStyle())
                }
                .padding(AppTokens.Spacing.lg)
                .background(AppTokens.Colors.surface)
                .cornerRadius(AppTokens.Shapes.large)
            }
            // 关键修改：统一容器内边距（仅横向+纵向统一16dp，无额外顶部8dp）
            // 与原 Kotlin Column 的 .padding(AppTokens.Spacing.lg) 完全对齐
            .padding(AppTokens.Spacing.lg)
        }
        .ignoresSafeArea(.all)
        .statusBarHidden(true) // 隐藏状态栏，避免额外留白
    }
}

@main
struct OrderTrackingApp: App {
    var body: some Scene {
        WindowGroup {
            RootScreen()
        }
    }
}

struct RootScreen_Previews: PreviewProvider {
    static var previews: some View {
        RootScreen()
    }
}

File: suc/kotlin/044.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Slider
import androidx.compose.material3.SliderDefaults
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableFloatStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "058*OrderTracking*en"
private const val UI_TYPE = "ECommerce"
private const val STYLE_THEME = "Retro Flat"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFFFF7043)
        val secondary = Color(0xFFFFB74D)
        val tertiary = Color(0xFFFFD180)
        val background = Color(0xFFFFF3E0)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFFFE0B2)
        val outline = Color(0xFFD7CCC8)
        val success = Color(0xFF66BB6A)
        val warning = Color(0xFFFFCA28)
        val error = Color(0xFFD32F2F)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFF3E2723)
        val onBackground = Color(0xFF3E2723)
        val onSurface = Color(0xFF3E2723)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.Bold)
        val title = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium)
    }
    object Shapes {
        val small = RoundedCornerShape(6.dp)
        val medium = RoundedCornerShape(10.dp)
        val large = RoundedCornerShape(16.dp)
    }
    object Spacing {
        val sm = 6.dp
        val md = 10.dp
        val lg = 16.dp
        val xl = 24.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(2.dp, 4.dp, 2.dp, 0.1f)
        val level2 = ShadowSpec(6.dp, 8.dp, 4.dp, 0.15f)
        val level3 = ShadowSpec(10.dp, 12.dp, 6.dp, 0.18f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

@Composable
fun RootScreen() {
    val progress = remember { mutableFloatStateOf(0.45f) }
    Scaffold(
        contentWindowInsets = androidx.compose.foundation.layout.WindowInsets(0),
        containerColor = AppTokens.Colors.background
    ) { pad ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(pad)
                .padding(AppTokens.Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
        ) {
            Text("Order Tracking", style = AppTokens.TypographyTokens.display, color = AppTokens.Colors.onBackground)
            Card(
                shape = AppTokens.Shapes.large,
                colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                elevation = CardDefaults.cardElevation(AppTokens.ElevationMapping.level2.elevation),
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(AppTokens.Spacing.lg),
                    verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
                ) {
                    Text("Order #25491", style = AppTokens.TypographyTokens.title, color = AppTokens.Colors.onSurface)
                    Text("Estimated Delivery: 2 days", style = AppTokens.TypographyTokens.body, color = AppTokens.Colors.secondary)
                    LinearProgressIndicator(
                        progress = { progress.floatValue },
                        modifier = Modifier.fillMaxWidth().height(10.dp),
                        color = AppTokens.Colors.primary,
                        trackColor = AppTokens.Colors.surfaceVariant
                    )
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Text("Processing", style = AppTokens.TypographyTokens.label, color = AppTokens.Colors.secondary)
                        Text("${(progress.floatValue * 100).toInt()}%", style = AppTokens.TypographyTokens.label, color = AppTokens.Colors.primary)
                    }
                }
            }
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(AppTokens.Colors.surface, AppTokens.Shapes.large)
                    .padding(AppTokens.Spacing.lg)
            ) {
                Column(
                    verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Text("Adjust Progress", style = AppTokens.TypographyTokens.title, color = AppTokens.Colors.onSurface)
                    Slider(
                        value = progress.floatValue,
                        onValueChange = { progress.floatValue = it },
                        colors = SliderDefaults.colors(
                            thumbColor = AppTokens.Colors.primary,
                            activeTrackColor = AppTokens.Colors.primary,
                            inactiveTrackColor = AppTokens.Colors.surfaceVariant
                        ),
                        modifier = Modifier.fillMaxWidth()
                    )
                    Button(
                        onClick = { progress.floatValue = 1f },
                        colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.primary),
                        shape = AppTokens.Shapes.medium,
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Text("Mark as Delivered", style = AppTokens.TypographyTokens.label, color = AppTokens.Colors.onPrimary)
                    }
                }
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFFFF3E0)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}
